src/components/common/sidebar.tsx

```diff
@@ -0,0 +1,75 @@
+"use client";
+
+import { useState } from "react";
+import Link from "next/link";
+import { Menu, X } from "lucide-react";
+import { navLists } from "@/constants";
+import Image from "next/image";
+
+const Sidebar = () => {
+  const [showSidebar, setShowSidebar] = useState(false);
+
+  const MenuItem = () => (
+    <ul className="flex flex-col gap-6 px-2 mt-4">
+      {navLists.map((nav) => (
+        <Link
+          key={nav.id}
+          href={nav.navigate_url}
+          onClick={() => setShowSidebar(false)}
+        >
+          <li className="text-white/60 hover:text-white">{nav.name}</li>
+        </Link>
+      ))}
+    </ul>
+  );
+
+  const MenuOverlay = () => (
+    <div
+      className="fixed top-0 bottom-0 left-0 right-0 z-30 flex md:hidden bg-black/40"
+      onClick={() => {
+        setShowSidebar(false);
+      }}
+    />
+  );
+
+  const MenuBarMobile = () => (

```

Comment:
better to move this code to a new component. create a navbar folder and move the MenuItem, MenuOverlay, and MenuBarMobile components into it.

Reply:
![image](https://github.com/user-attachments/assets/725f5526-bad4-4fb3-96d0-df9301aa4191)
Structure looks fine now?

Reply:
Or this one?

Reply:
looks good! approved @adzaky